apps/api/e2e/idempotency.e2e.ts

```diff
@@ -0,0 +1,141 @@
+import { UserSession } from '@novu/testing';
+import { expect } from 'chai';
+describe('Idempotency Test', async () => {
+  let session: UserSession;
+  const path = '/v1/testing/idempotency';
+  const idempotencyHeaderKey = 'Idempotency-Key';
+
+  describe('when enabled', () => {
+    before(async () => {
+      session = new UserSession();
+      await session.initialize();
+      process.env.API_IDEMPOTENCY_ENABLED = 'true';
+    });
+
+    it('should return cached same response for duplicate requests', async () => {
```

Comment:
I would add a test/ requirement that a request should return the idempotency key in the header also.

This way the key is sent to the system and back for the same request.

Reply:
done, i also changed the aproach a little bit, i found a bug while testing concurrent requests. solution is more performant as well now.